plugin/src/main/scala/com/typesafe/dbuild/plugin/StateHelpers.scala

```diff
@@ -1,40 +1,39 @@
 package com.typesafe.dbuild.plugin
 
 import sbt._
-import com.typesafe.dbuild.model
+import sbt.dbuild.hack.DbuildHack.ExceptionCategory
 import org.apache.commons.io.FileUtils.writeStringToFile
 
 object StateHelpers {
   def getProjectRefs(extracted: Extracted): Seq[ProjectRef] =
     extracted.structure.allProjectRefs
 
-  private def saveMsg(e: Throwable, lastMsgFile: File) = {
+  private def saveMsg(e: Throwable, lastMsgFile: File): Nothing = {
     val msg = e match {
-      case x:sbt.Incomplete =>
-        x.message match {
-          case Some(s) => s
-          case None => x.directCause match {
-            case Some(i) => Option(i.getMessage) getOrElse ""
-            case None => ""
-          }
-        }
+      case x: Incomplete =>
+        x.message orElse (x.directCause flatMap (t => Option(t.getMessage))) getOrElse ""
```

Comment:
This is a matter of style, but in case it can be reduced to the simpler:
```
x.message getOrElse (x.directCause map {_.getMessage} getOrElse "")
```

Reply:
The trouble with that encoding is if the direct cause's message is null you get null instead of the empty string, because Some#map always returns Some even if the value is null.

Reply:
True. I maintain that the original code was more readily understandable. :)